[BUG]: github_repository_environment CustomizeDiff panics on nil reviewers after upgrade from v6.11.x to v6.12.x

### Expected Behavior

Upgrading from `terraform-provider-github` v6.11.x to v6.12.x should work seamlessly for existing `github_repository_environment` resources, including those that have no `reviewers` block configured.

### Actual Behavior

`terraform plan` crashes with a nil pointer dereference panic in the new `CustomizeDiff` function when processing existing `github_repository_environment` resources that have no reviewers:

```
Stack trace from the terraform-provider-github_v6.12.1 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 203 [running]:
github.com/integrations/terraform-provider-github/v6/github.resourceGithubRepositoryEnvironmentDiff({0x1260dc8?, 0x20d058af4c90?}, 0x20d058930820?, {0xf42c40?, 0x20d0587b6e80?})
	github.com/integrations/terraform-provider-github/v6/github/resource_github_repository_environment.go:133 +0x1c5
github.com/integrations/terraform-provider-github/v6/github.resourceGithubRepositoryEnvironment.All.func3(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.2/helper/customdiff/compose.go:52 +0xd9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).PlanResourceChange(...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.38.2/helper/schema/grpc_provider.go:1133 +0xeff
```

### Terraform Version

```
Terraform v1.15.4
on linux_amd64
+ provider registry.terraform.io/integrations/github v6.12.1
```

### GitHub Installation Type

- [x] GitHub Enterprise Cloud with Personal Accounts (github.com)

### Affected Resource(s)

- `github_repository_environment`

### Terraform Configuration Files

```hcl
resource "github_repository_environment" "example" {
  repository  = "my-repo"
  environment = "staging"
  # no reviewers block — this is the trigger
}
```

Any `github_repository_environment` resource **without** a `reviewers` block that was created under v6.11.x triggers the panic when upgrading to v6.12.x.

### Steps to Reproduce

1. Create a `github_repository_environment` resource **without** a `reviewers` block using provider v6.11.x
2. Upgrade the provider to v6.12.0 or v6.12.1
3. Run `terraform plan`
4. The provider panics during `PlanResourceChange`

### Debug Output

**Root cause analysis:**

The `CustomizeDiff` function introduced in v6.12.0 ([PR #3162](https://github.com/integrations/terraform-provider-github/pull/3162)) does not handle the case where `reviewers` is empty or contains a nil element after state migration.

In [`resource_github_repository_environment.go` line 133](https://github.com/integrations/terraform-provider-github/blob/v6.12.1/github/resource_github_repository_environment.go#L133):

```go
o := v.([]any)[0]          // returns nil when reviewers list has a nil element
o.(map[string]any)["teams"] // panics: nil is not map[string]any
```

The `StateUpgrader` (SchemaVersion 0→1) only adds `repository_id` — it does not transform the `reviewers` attribute. When the SDK coerces the old state into the new schema (where `reviewers` changed from `MaxItems: 6` to `MaxItems: 1`), it can produce a list like `[nil]`, which the `CustomizeDiff` doesn't guard against.

Note that `expandReviewers` in the same file **does** have a proper nil check (`if m != nil`), but `CustomizeDiff` lacks it.

**Suggested fix** — add a nil guard before the type assertion:

```go
if v != nil {
    items := v.([]any)
    if len(items) > 0 && items[0] != nil {
        o := items[0].(map[string]any)
        // ... rest of the logic
    }
}
```

---

*This issue was filed with AI assistance (analysis and writeup). The bug, stack trace, and root cause were verified by a human engineer.*

--- a/github/resource_github_repository_environment.go
+++ b/github/resource_github_repository_environment.go
@@ -127,28 +127,44 @@ func resourceGithubRepositoryEnvironmentDiff(_ context.Context, d *schema.Resour
 		return nil
 	}
 
-	if v, ok := d.GetOk("reviewers"); ok {
-		if c, ok := v.([]any); ok && len(c) > 0 {
-			if o, ok := c[0].(map[string]any); ok {
-				count := 0
+	reviewersVal, ok := d.GetOk("reviewers")
+	if !ok {
+		return nil
+	}
 
-				if t, ok := o["teams"]; ok {
-					if s, ok := t.(*schema.Set); ok {
-						count += s.Len()
-					}
-				}
+	reviewersCol, ok := reviewersVal.([]any)
+	if !ok || len(reviewersCol) == 0 || reviewersCol[0] == nil {
+		return nil
+	}
 
-				if u, ok := o["users"]; ok {
-					if s, ok := u.(*schema.Set); ok {
-						count += s.Len()
-					}
-				}
+	reviewers, ok := reviewersCol[0].(map[string]any)
+	if !ok {
+		return nil
+	}
 
-				if count > 6 {
-					return fmt.Errorf("a maximum of 6 reviewers (users and teams combined) can be set for an environment")
-				}
-			}
+	teamsVal, teamsOk := reviewers["teams"]
+	usersVal, usersOk := reviewers["users"]
+
+	if !teamsOk && !usersOk {
+		return nil
+	}
+
+	reviewersCount := 0
+
+	if teamsOk {
+		if teamsCol, ok := teamsVal.(*schema.Set); ok {
+			reviewersCount += teamsCol.Len()
 		}
+	}
+
+	if usersOk {
+		if usersCol, ok := usersVal.(*schema.Set); ok {
+			reviewersCount += usersCol.Len()
+		}
+	}
+
+	if reviewersCount > 6 {
+		return fmt.Errorf("a maximum of 6 reviewers (users and teams combined) can be set for an environment")
 	}
 
 	return nil

--- a/github/resource_github_repository_environment.go
+++ b/github/resource_github_repository_environment.go
@@ -128,18 +128,26 @@ func resourceGithubRepositoryEnvironmentDiff(_ context.Context, d *schema.Resour
 	}
 
 	if v, ok := d.GetOk("reviewers"); ok {
-		count := 0
-		o := v.([]any)[0]
-		if t, ok := o.(map[string]any)["teams"]; ok {
-			count += t.(*schema.Set).Len()
-		}
+		if c, ok := v.([]any); ok && len(c) > 0 {
+			if o, ok := c[0].(map[string]any); ok {
+				count := 0
 
-		if t, ok := o.(map[string]any)["users"]; ok {
-			count += t.(*schema.Set).Len()
-		}
+				if t, ok := o["teams"]; ok {
+					if s, ok := t.(*schema.Set); ok {
+						count += s.Len()
+					}
+				}
 
-		if count > 6 {
-			return fmt.Errorf("a maximum of 6 reviewers (users and teams combined) can be set for an environment")
+				if u, ok := o["users"]; ok {
+					if s, ok := u.(*schema.Set); ok {
+						count += s.Len()
+					}
+				}
+
+				if count > 6 {
+					return fmt.Errorf("a maximum of 6 reviewers (users and teams combined) can be set for an environment")
+				}
+			}
 		}
 	}
 

--- a/github/resource_github_repository_environment_migration.go
+++ b/github/resource_github_repository_environment_migration.go
@@ -45,7 +45,7 @@ func resourceGithubRepositoryEnvironmentV0() *schema.Resource {
 			"reviewers": {
 				Type:        schema.TypeList,
 				Optional:    true,
-				MaxItems:    1,
+				MaxItems:    6,
 				Description: "The environment reviewers configuration.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
@@ -53,14 +53,12 @@ func resourceGithubRepositoryEnvironmentV0() *schema.Resource {
 							Type:        schema.TypeSet,
 							Elem:        &schema.Schema{Type: schema.TypeInt},
 							Optional:    true,
-							MaxItems:    6,
 							Description: "Up to 6 IDs for teams who may review jobs that reference the environment. Reviewers must have at least read access to the repository. Only one of the required reviewers needs to approve the job for it to proceed.",
 						},
 						"users": {
 							Type:        schema.TypeSet,
 							Elem:        &schema.Schema{Type: schema.TypeInt},
 							Optional:    true,
-							MaxItems:    6,
 							Description: "Up to 6 IDs for users who may review jobs that reference the environment. Reviewers must have at least read access to the repository. Only one of the required reviewers needs to approve the job for it to proceed.",
 						},
 					},

--- a/github/resource_github_repository_environment_test.go
+++ b/github/resource_github_repository_environment_test.go
@@ -153,12 +153,80 @@ resource "github_repository_environment" "test" {
 					Config: config,
 					ConfigStateChecks: []statecheck.StateCheck{
 						statecheck.ExpectKnownValue("github_repository_environment.test", tfjsonpath.New("repository_id"), knownvalue.NotNull()),
+						statecheck.ExpectKnownValue("github_repository_environment.test", tfjsonpath.New("reviewers"), knownvalue.ListSizeExact(1)),
 					},
 				},
 				{
 					Config: configUpdated,
 					ConfigStateChecks: []statecheck.StateCheck{
 						statecheck.ExpectKnownValue("github_repository_environment.test", tfjsonpath.New("reviewers"), knownvalue.ListSizeExact(0)),
+					},
+				},
+			},
+		})
+	})
+
+	t.Run("update_to_add_reviewers", func(t *testing.T) {
+		randomID := acctest.RandStringFromCharSet(5, acctest.CharSetAlphaNum)
+		repoName := fmt.Sprintf("%s%s", testResourcePrefix, randomID)
+		envName := "test"
+
+		preConfig := fmt.Sprintf(`
+resource "github_team" "test" {
+	name        = "%[1]s"
+	description = "test"
+	privacy     = "closed"
+}
+
+resource "github_repository" "test" {
+	name      = "%[1]s"
+	visibility = "public"
+}
+
+resource "github_team_repository" "test" {
+	team_id    = github_team.test.id
+	repository = github_repository.test.name
+	permission = "pull"
+}
+`, repoName)
+
+		config := fmt.Sprintf(`
+%s
+
+resource "github_repository_environment" "test" {
+	repository  = github_repository.test.name
+	environment = "%s"
+}
+`, preConfig, envName)
+
+		configUpdated := fmt.Sprintf(`
+%s
+
+resource "github_repository_environment" "test" {
+	repository  = github_repository.test.name
+	environment = "%s"
+
+	reviewers {
+		teams = [github_team_repository.test.team_id]
+	}
+}
+`, preConfig, envName)
+
+		resource.Test(t, resource.TestCase{
+			PreCheck:          func() { skipUnlessHasOrgs(t) },
+			ProviderFactories: providerFactories,
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					ConfigStateChecks: []statecheck.StateCheck{
+						statecheck.ExpectKnownValue("github_repository_environment.test", tfjsonpath.New("repository_id"), knownvalue.NotNull()),
+						statecheck.ExpectKnownValue("github_repository_environment.test", tfjsonpath.New("reviewers"), knownvalue.ListSizeExact(0)),
+					},
+				},
+				{
+					Config: configUpdated,
+					ConfigStateChecks: []statecheck.StateCheck{
+						statecheck.ExpectKnownValue("github_repository_environment.test", tfjsonpath.New("reviewers"), knownvalue.ListSizeExact(1)),
 					},
 				},
 			},

